test/pylib/artifact_registry.py

```diff
@@ -23,30 +23,42 @@ class ArtifactRegistry:
     reprsented in the artifact registry. """
 
     def __init__(self) -> None:
-        self.suite_artifacts: Dict[Suite, List[Artifact]] = {}
-        self.exit_artifacts: List[Artifact] = []
+        self.suite_artifacts: Dict[str, List[Artifact]] = {}
+        self.exit_artifacts: Dict[str, List[Artifact]] = {}
```

Comment:
Here you can use `Optional[Suite]` as the key type. In `add_exit_artifact` you can then use `suite` directly as the key. No need to use a special `__exit__` string.

Reply:
ok